### PR TITLE
Migrate to concordium-dapp-libraries 

### PR DESCRIPTION
**Pull request by bisgardo**
Replace adhoc implementations of Browser Wallet/WalletConnect integration with the published libraries.

This mostly deletes code related to connection handling that the dapp library now takes care of in a component and a couple of hooks.

Care has been taken to ensure that the migration introduces as few functional changes as possible. In particular, it keeps the behavior that inactive connectors stay alive by using `persistentConnectorType`. This serves to illustrate that this functionality basically comes for free. Periodic pinging of WalletConnect is included as well.

A few "unrelated" migrations of SDK types are also included.

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -44,221 +60,52 @@ export default function App() {
...
-    const canSmash = useMemo(() => {
-        if (wallet === 'browserwallet' && contract) {
-            return browserwalletConnectedAccount === contract.owner.address;
-        }
-        if (wallet === 'walletconnect2' && walletconnect2ConnectedSession && contract) {
-            return resolveAccount(walletconnect2ConnectedSession) === contract.owner.address;
-        }
-        return false;
-    }, [wallet, browserwalletConnectedAccount, walletconnect2ConnectedSession, contract]);
+    // TODO Need an interface ('canSmash', 'handleSubmitDeposit', etc.).
```
Comment (shjortConcordium) [inline src/App.tsx:63]:
```suggestion
```
I guess the interface is the Piggybank Props. (Don't see the point of this TODO :thinking: )

File: src/Contract.tsx [hunk 2]
```diff
@@ -156,7 +156,7 @@ export function ContractManager(props: ModalProps) {
                                     </Row>
                                     <Row>
                                         <Col sm={2}>Balance:</Col>
-                                        <Col sm={10}>{currentContract.amount.microGtuAmount.toString()} μCCD</Col>
+                                        <Col sm={10}>{currentContract.amount.toString()} μCCD</Col>
```
Comment (shjortConcordium) [inline src/Contract.tsx:159]:
```suggestion
                                        <Col sm={10}>{currentContract.amount.toJSON()} μCCD</Col>
```
The toString() just returns the good old [object Object]
(Or do `currentContract.amount.microCcdAmount.toString()` )

**Review by shjortConcordium — APPROVED** [review]

File: src/Contract.tsx (hunk 2 above)
Comment (bisgardo) [reply src/Contract.tsx:159]: Dang I was certain that this had been implemented as `toString`. Nice catch, will revert.

File: src/App.tsx (hunk 1 above)
Comment (bisgardo) [reply src/App.tsx:63]: Addressed in followup: https://github.com/Concordium/concordium-dapp-piggybank/pull/4

**Pull request merged**